### PR TITLE
Add vuid 06309

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Last of #3292

### CODE REVIEW

File: layers/shader_validation.cpp [hunk 1]
```diff
@@ -2137,32 +2139,75 @@ bool CoreChecks::ValidateDecorations(SHADER_MODULE_STATE const* module) const {
...
+                bool found = false;
+                for (auto &bds : buffer_data_sizes) {
+                    if (bds.first == xfb_buffer.word(1)) {
+                        bds.second = std::max(bds.second, buffer_data_size);
+                        found = true;
+                        break;
+                    }
+                }
+                if (!found) {
+                    buffer_data_sizes.push_back({xfb_buffer.word(1), buffer_data_size});
```
Comment (ncesario-lunarg) [inline layers/shader_validation.cpp:2176]: `emplace_back`?

File: tests/vklayertests_pipeline_shader.cpp [hunk 2]
```diff
@@ -14079,6 +14079,79 @@ TEST_F(VkLayerTest, TestRuntimeSpirvTransformFeedback) {
         };
         CreatePipelineHelper::OneshotTest(*this, set_info, kErrorBit, "VUID-RuntimeSpirv-Stream-06312");
     }
+
+    {
+        uint32_t offset = transform_feedback_props.maxTransformFeedbackBufferDataSize / 2;
+        uint32_t count = transform_feedback_props.maxTransformFeedbackStreamDataSize / offset + 1;
+        if (count < 25) {
```
Comment (ncesario-lunarg) [inline tests/vklayertests_pipeline_shader.cpp:14086]: Is the 25 limit for letter-only variable names?

**Review by ncesario-lunarg — APPROVED** [review]
This LGTM, though the wording of the VUID is a little hard for me to follow (my brain isn't exactly working right now, though).

FYI @sfricke-samsung.

File: tests/vklayertests_pipeline_shader.cpp (hunk 2 above)
Comment (ziga-lunarg) [reply tests/vklayertests_pipeline_shader.cpp:14086]: The way I wrote the test with `char v = 'a' + i;`, I didn't want to get errors. I can fix this if needed.

Comment (ncesario-lunarg) [reply tests/vklayertests_pipeline_shader.cpp:14086]: 👍 do these need to be letters? e.g., can the identifiers be `%var1`, `%var2`, ...?

Comment (ncesario-lunarg) [reply tests/vklayertests_pipeline_shader.cpp:14086]: I don't think it needs to be changed, more just me asking for my understanding.

**Review by sfricke-samsung — APPROVED** [review]
So easily one of the most complex VUID statements in the spec. No TransformFeedback expert, but seems right... I imagine if anyone will actually be hitting this it would be an GL-to-VK layer like ANGLE

The changes we made from the last transform feedback PR for `ValidateDecorations()` work well here

File: tests/vklayertests_pipeline_shader.cpp (hunk 2 above)
Comment (sfricke-samsung) [reply tests/vklayertests_pipeline_shader.cpp:14086]: @ncesario-lunarg not sure why there is a concern for a letter limit

@ziga-lunarg I think just adding a comment on "why" 25 was picked above should be enough for me

**Pull request merged**